src/meta/src/hummock/hummock_manager.rs

```diff
@@ -269,17 +271,26 @@ where
         meta_store: &S,
         mut trx: Transaction,
         context_id: Option<HummockContextId>,
+        info: MetaLeaderInfo,
     ) -> Result<()> {
         if let Some(context_id) = context_id {
             if context_id == META_NODE_ID {
                 // Using the preserved meta id is allowed.
-            } else if let Some(worker) = self.cluster_manager.get_worker_by_id(context_id).await {
-                trx.check_exists(Worker::cf_name(), worker.key()?.encode_to_vec());
-            } else {
+            } else if self
+                .cluster_manager
+                .get_worker_by_id(context_id)
+                .await
+                .is_none()
+            {
                 // The worker is not found in cluster.
                 return Err(Error::InvalidContext(context_id));
             }
         }
+        trx.check_equal(
+            META_CF_NAME.to_owned(),
+            META_LEADER_KEY.as_bytes().to_vec(),
+            info.encode_to_vec(),
+        );
         meta_store.txn(trx).await.map_err(Into::into)
```

Comment:
What if context_id becomes invalid just before `meta_store.txn`?

Reply:
Good catch.... It seems that I need a memory lock for `context_id`, does it?

Reply:
> Good catch.... It seems that I need a memory lock for `context_id`, does it?

Yeah something prevents `ClusterManager` from `delete_worker_node`.